plugin.video.youtube/resources/lib/youtube/helper/subtitles.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+import xbmcvfs
+import re
+from resources.lib.kodion import simple_requests as requests
+
+
+class Subtitles(object):
+    LANG_NONE = 0
+    LANG_ALL = 1
+    LANG_CURR = 2
+    LANG_CURR_EN = 3
+    LANG_EN = 4
+    LANG_PROMPT = 5
+
+    SUBTITLE_LIST_URL = 'http://www.youtube.com/api/timedtext?type=list&v=%s'
+    SUBTITLE_URL = 'http://www.youtube.com/api/timedtext?fmt=vtt&v=%s&lang=%s'
+    SRT_FILE = 'special://temp/%s.%s.srt'
+
+    def __init__(self, context, video_id):
+        self.context = context
+        self.video_id = video_id
+        self.language = context.get_settings().get_string('youtube.language', 'en_US').replace('_', '-')
+        self.headers = {'Host': 'www.youtube.com',
+                        'Connection': 'keep-alive',
+                        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 '
+                                      '(KHTML, like Gecko) Chrome/53.0.2785.143 Safari/537.36',
+                        'Accept': '*/*',
+                        'DNT': '1',
+                        'Referer': 'https://www.youtube.com/tv',
+                        'Accept-Encoding': 'gzip, deflate',
+                        'Accept-Language': 'en-US,en;q=0.8,de;q=0.6'}
+
+    def get(self):
+        if self._languages() != self.LANG_NONE:
+            self.context.log_debug('Getting subtitles for video_id: %s, language_enum: %s' %
+                                   (self.video_id, self._languages()))
+            return self._get_subtitles()
+        else:
+            return []
+
+    def srt_filename(self, sub_language):
+        return self.SRT_FILE % (self.video_id, sub_language)
+
+    def subtitle_url(self, sub_language):
+        return self.SUBTITLE_URL % (self.video_id, sub_language)
+
+    def _languages(self):
+        return self.context.get_settings().subtitle_languages()
+
+    def _write_file(self, _file, contents):
+        self.context.log_debug('Writing subtitle file: %s' % _file)
+        f = xbmcvfs.File(_file, 'w')
```

Comment:
oh, sorry, i think youre correct. sorry for the noise. :) some try-except error handling would still be nice though when dealing with files.